src/main/java/net/snowflake/client/core/SFException.java

```diff
@@ -36,6 +36,18 @@ public SFException(ErrorCode errorCode, Object... params) {
     this.params = params;
   }
 
+  public SFException(String queryID, ErrorCode errorCode, Object... params) {
```

Comment:
after adding new constructors - can we mark some existing constructors as deprecated when are not used?